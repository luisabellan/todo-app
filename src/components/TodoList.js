import React from "react";
import Todo from "./Todo";

const TodoList = (props) => {
  return (
    <div className="todolist-wrapper">
      <ul data-testid="todos-ul">
        {props.todoItems.map((item) => (
          <Todo key={item.id} item={item} toggleItem={props.toggleItem} />
        ))}
      </ul>
    </div>
  );
};

export default TodoList;
