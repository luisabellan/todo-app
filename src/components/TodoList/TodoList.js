import React from 'react';
import Todo from '../Todo/Todo';



function TodoList(props) {


  return (
    <div className="todolist-wrapper" data-testid="todos" >
      {
        props.todos.map((todo) => (
          <Todo
            key={todo.id}
            data-testid={`checkbox ${todo.note}  `}
            handleChanges={props.handleChanges}
            todo={todo}
            toggleItem={props.toggleItem}
          />
        ))
      }
    </div >
  );
}

export default TodoList;
