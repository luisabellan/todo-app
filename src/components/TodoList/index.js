import React from 'react';
import Todo from '../Todo';
import PropTypes, { bool, number, string } from 'prop-types';

TodoList.propTypes = {
  todo: PropTypes.shape(
    {
      id: number,
      name: string,
      completed: bool
    }
  ),
  todos: PropTypes.arrayOf(PropTypes.object),
  toggleItem: PropTypes.func,
  handleChanges: PropTypes.func
};



function TodoList(props) {
  return (
    <div className="todolist-wrapper">
      <ul data-testid="todos-ul">
        {props.todos.map((todo) => (
          <Todo key={todo.id} handleChanges={props.handleChanges} todo={todo} toggleItem={props.toggleItem} />
        ))}
      </ul>
    </div>
  );
}

export default TodoList;
