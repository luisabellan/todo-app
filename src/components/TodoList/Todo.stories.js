import React from 'react';

import TodoList from './TodoList';

export default {
  component: TodoList,
  title: 'TodoList',
};

const Template = args => <TodoList {...args} />;

export const Default = Template.bind({});
Default.args = {
  todos: []

};

/* export const Undone = Template.bind({});
Undone.args = {
  todo: {
    ...Default.args.todo,
    completed: false,
  },
}; */
