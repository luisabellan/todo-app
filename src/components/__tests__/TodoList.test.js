import React from 'react';
import '@testing-library/jest-dom';

import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import TodoList from '../TodoList/TodoList';

const todos = [
  {
    id: "checkbox buy bread",
    name: 'buy bread',
    completed: false
  },
  {
    id: "checkbox buy milk",
    name: 'buy milk',
    completed: false
  },
  {
    id: "checkbox write blog",
    name: 'write blog',
    completed: false
  }
];

describe('todo list test', () => {
  afterEach(cleanup);
  it('matches snapshot', () => {
    const { asFragment } = render(<TodoList todos={todos} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should show name of todos', () => {
    render(<TodoList todos={todos} />);
    todos.forEach((d) => expect(screen.getByText(d.name)).toBeInTheDocument());
  });
});
