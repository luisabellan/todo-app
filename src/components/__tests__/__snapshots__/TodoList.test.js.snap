// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo list test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todolist-wrapper"
  >
    <ul
      data-testid="todos-ul"
    >
      <div
        class="item "
      >
        <p
          style="font-family: Indie Flower; font-size: 2rem;"
        >
          buy bread
        </p>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
